### PR TITLE
WIP: Resolve json rpc

### PR DESCRIPTION
**Pull request by lothropgauss**

### CODE REVIEW

File: server/jsonrpc_service.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	res, err := InternalResolve(args.Data, t.DB)
+	*result = res
+	return err
+}
+
+func (s *Server) StartJsonRPC111() error {
+	server := new(JSONServer)
+	server.DB = s.DB
+
+	port := ":8080"
```
Comment (moodyjon) [inline server/jsonrpc_service.go]: JSON port number should be a new thing in args.go, defaulting to 50001.

Then pass the args struct (or json parts of it) into StartJsonRPC111().

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+grpcio-tools==1.47.0
+idna==3.3
+protobuf==3.20.1
+pycparser==2.21
+PyJWT==2.4.0
+python-dateutil==2.8.2
+requests==2.28.1
+six==1.16.0
+uritemplate==4.1.1
+urllib3==1.26.11
```
Comment (moodyjon) [inline requirements.txt:16]: Is this needed by scripts/version.py? Did it get created inadvertently?

File: test_rpc.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+curl --request POST 'localhost:8686/' \
```
Comment (moodyjon) [inline test_rpc.sh]: Where does this port number come from? I would expect to see 8080 (or 50001)

Also does this need "/rpc" to match line 45 of jsonrpc_service.go?

I haven't been able to determine whether (Python) herald uses "/rpc" or any other path(s).

**Review by moodyjon — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (lothropgauss) [reply requirements.txt:16]: Yes, this is needed by the version.py, it was intentional.

File: server/jsonrpc_service.go (hunk 1 above)
Comment (lothropgauss) [reply server/jsonrpc_service.go]: Yup, I just added that. It's already carried in through the Server struct.

File: test_rpc.sh (hunk 3 above)
Comment (lothropgauss) [reply test_rpc.sh]: This is an old script added in by mistake, I'll move and change it.

**Review by moodyjon — APPROVED** [review]

**Pull request merged**